tests/native/test_models.py

```diff
@@ -262,6 +263,39 @@ def strlen_model(state):
         # Assert that every expected stdout has a matching output
         self.assertEqual(expected, stdouts)
 
+    def test_symbolic_fork_unique_solution(self):
+        # This binary is compiled using gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
+        # with flags: -g -static -fno-builtin
+        BIN_PATH = os.path.join(
+            os.path.dirname(__file__), "binaries/str_model_tests", "sym_strlen_test"
+        )
+        tmp_dir = tempfile.TemporaryDirectory(prefix="mcore_test_sym_")
+        m = Manticore(BIN_PATH, stdin_size=10, workspace_url=str(tmp_dir.name))
+
+        addr_of_strlen = 0x04404D0
+
+        @m.hook(addr_of_strlen)
+        def strlen_model(state):
+            stdin_1 = ArrayProxy(array=state.constraints.get_variable("STDIN")).select(1)
+            state.constrain(stdin_1 == 0)
```

Comment:
This constrains the second symbolic byte of the input to be zero, causing Manticore to fork on an expression that has only one possible solution.

Reply:
I'm pretty sure I didn't know you could use `ArrayProxy` to access a specific byte of `stdin` like this. Great find!